src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -389,10 +389,9 @@ private void RemoveFromStats(Stats current)
             {
                 for (int i = 0; i < _allStats.Count; i++)
                 {
-                    if (_allStats[i].TryGetTarget(out Stats? stats) && stats == current)
+                    if (_allStats[i].TryGetTarget(out Stats? stats))
                     {
                         _allStats.RemoveAt(i);
-                        break;
```

Comment:
I've removed the `break` as a defensive mechanism. If the code is bug-free, each `RemoveFromStats` should remove a single element from `_allStats`.

Reply:
> it could also remove elements from threads that recently died too.

Only if the element was not at `i+1`th position. `RemoveAt` is going to move it to `i`th position and the next iteration is going to skip it.

Reply:
Good catch. No strong opinion about that behavior. Maybe I should just comment what is the expected behavior?

Reply:
You could `i--;` after `RemoveAt(i);` to not miss it.